mission/functions/systems/ranks/fn_unit_next_rank.sqf

```diff
@@ -23,14 +23,20 @@ params
 	["_unit",objNull,[objNull]] // 0: OBJECT - player unit object
 ];
 
-private _data = [];
-{
-	//code
+private _rankConfig = getArray(missionConfigFile >> "gamemode" >> "rank" >> "ranks");
+
+// if maxrank UI will show an error if we try to do anything further
+private _maxRankPoints = selectMax (_rankConfig apply {(_x select 0) select 2});
+
+if (_unit getVariable ["vn_mf_db_rank",0] > _maxRankPoints) exitWith {
+	((_rankConfig select -1) select 0)
+};
```

Comment:
fix a bug when we're already at max rank -- show the players current rank when no more ranks to achieve.